cmd/kubelet/app/server.go

```diff
@@ -367,6 +373,32 @@ func UnsecuredDependencies(s *options.KubeletServer, featureGate featuregate.Fea
 	if err != nil {
 		return nil, err
 	}
+	var tp oteltrace.TracerProvider
+	if utilfeature.DefaultFeatureGate.Enabled(features.KubeletTracing) {
+		hostname, err := nodeutil.GetHostname(s.HostnameOverride)
+		if err != nil {
+			return nil, err
+		}
+		opts := []otlpgrpc.Option{}
+		resourceOpts := []otelsdkresource.Option{
+			otelsdkresource.WithAttributes(
+				semconv.ServiceNameKey.String(hostname),
+				semconv.ServiceInstanceIDKey.String(componentKubelet),
+			),
+		}
+		if s.KubeletConfiguration.Tracing.Endpoint != nil {
+			opts = append(opts, otlpgrpc.WithEndpoint(*s.KubeletConfiguration.Tracing.Endpoint))
+		}
+		opts = append(opts, otlpgrpc.WithInsecure())
+		tc := &configv1alpha1.TracingConfiguration{}
+		if err := configv1alpha1.Convert_config_TracingConfiguration_To_v1alpha1_TracingConfiguration(&s.KubeletConfiguration.Tracing, tc, nil); err != nil {
+			return nil, err
+		}
+		tp, err = traces.InitTracing(context.Background(), *tc, opts, resourceOpts)
+		if err != nil {
+			return nil, err
```

Comment:
Same comment as above re: returning an errors in this block. I wonder if it would make sense to factor this all out into a helper function, as this function previously didn't do very much. The logic for the other plugins isn't included here.

Reply:
agreed, thanks, updated